### DISCUSSION TITLE
Human language definition for drop-shadows?

### DISCUSSION THREAD

**Discussion by BillRaymond** [ImageMagick/ImageMagick#5356] [Help]
I am using ImageMagick to create featured images for each of my podcast posts, like the guest image you see below:
![](https://agileinaction.com/uploads/2022-06-07-building-a-culture-of-psychological-safety-no-play-icon.png)

In my Alpine UNIX shell command, using ImageMagick 7.1.0-35, I create a drop-shadow for each guest, like this (which you can also see in the picture above):

```
magick convert guest.png \( +clone -background black -shadow 60x5+6+6 \) \
    +swap -background none -layers merge +repage guest.png
```

So that works and all, but I had to track it down on here and SO to figure it out. Looking at the docs [here](https://imagemagick.org/script/command-line-options.php#shadow), I get this definition:

```
-shadow percent-opacity{xsigma}{+-}x{+-}y{%}
Simulate an image shadow.
```

Looking around on the ImageMagick site and Google searches, I cannot find a good definition for what `xsigma` or any of those other elements are. Here is what I think:

Given:
```
-shadow 60x5+6+6
```

I assume `+6+6` is how right to the right and bottom the pixels go to make the shadow? But that is just an assumption. I have no idea why I even have 60x5, but since it works, I use it :-)

My effect in Adobe Illustrator looks like this:
```
Mode: Multiply
Opacity: 75%
x-offset: 6 px
y-offset: 6 px
Blur: 5 px
```
So it is hard for me to translate in my head how ImageMagick maps similar settings (or if there is even a 1:1 similarity to those settings).

Could someone please define the settings for `-shadow` in more detail than the current docs?

Even better, if you can walk me through the following line of code I am using (but do not understand), it would be greatly appreciated!

```
magick convert guest.png \( +clone -background black -shadow 60x5+6+6 \) \
    +swap -background none -layers merge +repage guest.png
```

Thanks!

**Comment by fmw42** [comment]
>  -shadow percent-opacity{xsigma}{+-}x{+-}y{%} 

sigma is the blur. X and Y are the offsets or distance which can be specified in pixels or %

**Reply by BillRaymond** [reply]
Thank you! How do you define the blur? For example, mine is `60x5`. Could you explain what that means?

**Reply by fmw42** [reply]
blur is the sigma of a normal distribution. It is similar to standard deviation. It is a measure of the width of the normal curve produced by blurring. It is approximately radius/3 (r=3*sigma), where radius is the width of the normal curve. See the graph in the standard deviation section of https://en.wikipedia.org/wiki/Normal_distribution.  How does Photoshop define blur?

**Reply by BillRaymond** [reply]
Thank you for sharing that detail. Although I will say it is still a little confusing how to take in what you just shared with me. I think you may be explaining what blur is rather than what the settings are in ImageMagick? You ask how Photoshop defines blur. It looks like the image below:

<img width="1125" alt="photoshop-shadow" src="https://user-images.githubusercontent.com/15895828/180543791-217335ea-67bc-4f80-a256-630e257478bd.png">

Here are basic definitions as I understand them:

* `blend mode` defines how the shadow will appear against the backdrop. The `multiply` setting is the equivalent of a shadow appearing over a transparent layer
* `opacity` defines how prominent the shadow is. A lower number makes the shadow more transparent. I set the value to `100%`, which means the shadow will be as dark as possible
* `angle` defines how to display the shadow. I set it to `120`, which in this case casts the shadow to the bottom-right of the box
* `distance` is how far away the shadow is from the box. Pretend there is a book on your desk, casting a tight shadow next to it. Then pick the book up, the shadow casts further over the desk and appear further away from the book. The lower the number, the more tightly the shadow appears attached to the box
* `spread` is how fuzzy the shadow is. In this case, if I pumped the number uwould 100%, it will look like a black box under the red box because there will be no fuzziness at all
* `size` is the size of the spread, so if I push that number up to, say, 250 px, the red box would be almost entirely surrounded by a shadow, but with the most prominent parts of it at the bottom-right, as defined by the `angle`
* `contour` is a bit of a mystery to me. What I know is if you select what I have in the image with a straight line (bottom-left to top-right), you get a standard shadow. Most others make the shadows look like neon lights and I have no idea why :-). I do not _think_ this is a correlation to the bell curve you mention in the article, but perhaps it is

It would be cool to know how the PhotoShop or my previous Illustrator example I shared aligns with ImageMagick's features. But more important than that is to understand ImageMagick so I can visualize what those settings mean.
 
So far, from this thread, I _think_ what I got from this is as follows:

**My blur settings**
```
-shadow 60x5+6+6
```

**My interpretation of those settings**
1. `-shadow` Cast a shadow on the object
2. `60` Make that shadow 60% visible (like the PhotoShop `opacity` option)
3. `60x5` Write `60x5` for some reason 🤷?
4. `+6` Given the `gravity` setting (which mine is at `center`, the shadow should not case more than `+6 pixels` to the right of the image
5. `+6` Given the `gravity` setting (which mine is at `center`, the shadow should not case more than `+6 pixels` to the bottom of the image

**Reply by fmw42** [reply]
percent opacity == PS opacity
x,y == PS distance in pixels
sigma == PS spread. (though spread may correspond more to radius = 3xsigma)

== means corresponds, not exactly the same values. I have not made a one-to-one comparison.

**Comment by snibgo** [comment]
@BillRaymond: As you have said, the syntax is:

> -shadow percent-opacity{xsigma}{+-}x{+-}y{%}

So in your...
```
-shadow 60x5+6+6
```
... the first number, "60", is "percent-opacity". I would guess that is the same as Adobe Illustrator's "Opacity".

**Reply by BillRaymond** [reply]
Ah! That is useful to know, thank you! Can you share what is the `x5` in the `60x5`?

**Comment by snibgo** [comment]
The "x" is just to separate the numbers. "5" is sigma, as fmw42 explained above.

**Reply by BillRaymond** [reply]
Thank you all for the help. I am off and running!
